packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -235,7 +237,7 @@ export function useAsyncData<
     if (getCurrentInstance()) {
       onServerPrefetch(() => promise)
     } else {
-      nuxt.hook('app:created', () => promise)
+      nuxt.hook('app:created', async () => { await promise })
```

Comment:
@darioferderber I'm curious about the reason behind the modification made to this line of code. Could you please explain it?

Reply:
Oh I got it! because type defined in here `Promise<void>`

https://github.com/nuxt/nuxt/blob/5bb6a13eaeb5a283f0021b2144824ab1459ab82e/packages/nuxt/src/app/nuxt.ts#L25-L29